core/src/main/java/org/apache/gravitino/EntityStoreV2.java

```diff
@@ -0,0 +1,219 @@
+      throws NoSuchEntityException, IOException;
+
+  /**
+   * Delete the entity from the underlying storage by the specified {@link
+   * org.apache.gravitino.NameIdentifier}.
+   *
+   * @param ident the name identifier of the entity
+   * @param entityType the type of the entity to be deleted
+   * @return true if the entity exists and is deleted successfully, false otherwise
+   * @throws IOException if the delete operation fails
+   */
+  default boolean delete(NameIdentifier ident, EntityType entityType) throws IOException {
+    return delete(ident, entityType, false);
+  }
+
+  /**
+   * Delete the entity from the underlying storage by the specified {@link
+   * org.apache.gravitino.NameIdentifier}.
+   *
+   * @param ident the name identifier of the entity
+   * @param entityType the type of the entity to be deleted
+   * @param cascade support cascade delete or not
+   * @return true if the entity exists and is deleted successfully, false otherwise
+   * @throws IOException if the delete operation fails
+   */
+  boolean delete(NameIdentifier ident, EntityType entityType, boolean cascade) throws IOException;
+
+  /**
+   * List all the entities with the specified {@link org.apache.gravitino.NameIdentifier}, and
+   * deserialize them into the specified {@link Entity} object.
+   *
+   * <p>Note. Depends on the isolation levels provided by the underlying storage, the returned list
+   * may not be consistent.
+   *
+   * @param ident the name identifier of the entity
+   * @param <E> class of the entity
+   * @param type the detailed type of the entity
+   * @param entityType the general type of the entity
+   * @param relationType the relation type of entities to be list out with the specified entity
+   * @throws IOException if the list operation fails
+   * @return the list of entities
+   */
+  <E extends Entity & HasIdentifier> List<E> listEntitiesByRelation(
+      NameIdentifier ident, Class<E> type, EntityType entityType, RelationType relationType)
+      throws IOException;
+
+  /**
+   * @param start NameIdentifier of the starting point of the relationship.
+   * @param type EntityType of the starting point.
+   * @param relationType RelationType of the relationship.
+   * @return List of MetadataObject that are related to the starting point.
+   * @throws IOException if the list operation fails.
+   */
+  List<MetadataObject> listMetadataObjectsByRelation(
+      NameIdentifier start, EntityType type, RelationType relationType) throws IOException;
+
+  /**
+   * Add a relation between two entities.
+   *
+   * @param start NameIdentifier of the starting point of the relationship.
+   * @param startType EntityType of the starting point.
+   * @param end NameIdentifier of the ending point of the relationship.
+   * @param endType EntityType of the ending point.
+   * @param type RelationType of the relationship.
+   * @throws AlreadyExistsException if the relation already exists.
+   */
+  void addRelation(
```

Comment:
Do we support override relation?

Reply:
I analyzed the current call chain and it doesn't seem necessary to add an update interface.  In order to make it more extensive, I agree at it's more preferable to add an `update` funtionity.